src/pages/Map/Map.js

```diff
@@ -0,0 +1,180 @@
+import { MapContainer, TileLayer, Marker, Popup } from 'react-leaflet';
+import { useRef } from 'react';
+import styled from 'styled-components';
+import { FaCheck as Create } from 'react-icons/fa';
+import { IoMdArrowRoundBack as ArrowBack } from 'react-icons/io';
+import 'leaflet/dist/leaflet.css';
+import './Map.css';
+import ScreenReaderOnly from '../../components/ScreenReaderOnly';
+import L from 'leaflet';
+import osm from './osm-provider';
+import MapboxGeocoder from '@mapbox/mapbox-gl-geocoder';
+import mapboxgl from 'mapbox-gl';
+import { useEffect } from 'react';
+import { useLocalStorage } from 'usehooks-ts';
+import { Link } from 'react-router-dom';
+
+export default function BasicMap({ onNewDiveClick, diveData, viewPort }) {
+  mapboxgl.accessToken = process.env.REACT_APP_ACCESSTOKEN;
+  const markerIcon = new L.Icon({
+    iconUrl: require('./icon.png'),
+    iconSize: [32, 32],
+    iconAnchor: [16, 32],
+  });
+  const markerIconRed = new L.Icon({
+    iconUrl: require('./iconRed.png'),
+    iconSize: [32, 32],
+    iconAnchor: [16, 32],
+  });
+  const [destinationMapbox, setDestinationMapbox] = useLocalStorage(
+    'destinationMapBox',
+    [52.51667, 13.38333]
+  );
+  const ZOOM_LEVEL = 8;
+  const mapRef = useRef();
+  const [center, setCenter] = useLocalStorage('center', {
+    lat: destinationMapbox[0],
+    lng: destinationMapbox[1],
+  });
+  const position = [destinationMapbox[0], destinationMapbox[1]];
+  const divesWithCoordinates = diveData.filter(
+    dive => dive.coordinates[0] !== null
+  );
+
+  function handleDiveViewPort(viewPort) {
+    if (viewPort !== '') {
+      return (
+        setCenter({ lat: viewPort[0], lng: viewPort[1] }),
+        window.location.reload()
+      );
+    } else {
+      return;
+    }
+  }
+
+  useEffect(() => {
+    const geocoderDestination = new MapboxGeocoder({
+      accessToken: mapboxgl.accessToken,
+      types: 'country, region, place, poi',
+      limit: 5,
+      placeholder: 'e.g. Lissabon',
+      minLength: 2,
+    });
+    handleDiveViewPort(viewPort);
+    geocoderDestination.on('result', e => {
+      return setDestinationMapbox([
+        e.result.center[1],
+        e.result.center[0],
+        e.result,
+      ]);
+    });
+    geocoderDestination.addTo('#geocoderdestination');
+  }, []);
+
+  return (
+    <>
+      <MapContainer center={center} zoom={ZOOM_LEVEL} ref={mapRef} id="map">
+        <TileLayer
+          url={osm.maptiler.url}
+          attribution={osm.maptiler.attribution}
+        />
+        <Marker position={position} icon={markerIconRed}>
+          <Popup>
+            Add new Dive
+            <StyledLink
+              to="/divelog"
+              aria-label="searchTags"
```

Comment:
```suggestion
              aria-label="go-to-dive-logs"
```
something like this?